### ISSUE TITLE
[helm3] Kubernetes cluster unreachable

### ISSUE THREAD

**Issue by mungaij83** [helm/helm#7641]
I have been working with helm2 for quite some time and it had a flag to allow tls over any certificate using **--insecure-skip-tls-verify** flag. I am upgrading to helm3 and I faced this issue which I think its an issue with tls verification.

$ helm ls -v=10 --debug
```
...
I0220 10:46:44.480718   32300 round_trippers.go:443] GET https://10.2.0.12:8433/version?timeout=32s  in 30009 milliseconds
I0220 10:46:44.480821   32300 round_trippers.go:449] Response Headers:
Error: Kubernetes cluster unreachable
helm.go:75: [debug] Kubernetes cluster unreachable
helm.sh/helm/v3/pkg/kube.(*Client).IsReachable
        /home/circleci/helm.sh/helm/pkg/kube/client.go:91
helm.sh/helm/v3/pkg/action.(*List).Run
        /home/circleci/helm.sh/helm/pkg/action/list.go:141
main.newListCmd.func1
        /home/circleci/helm.sh/helm/cmd/helm/list.go:80
github.com/spf13/cobra.(*Command).execute
        /go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:826
github.com/spf13/cobra.(*Command).ExecuteC
        /go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:914
github.com/spf13/cobra.(*Command).Execute
        /go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:864
main.main
        /home/circleci/helm.sh/helm/cmd/helm/helm.go:74
runtime.main
        /usr/local/go/src/runtime/proc.go:203
runtime.goexit
        /usr/local/go/src/runtime/asm_amd64.s:1357
```

Is it possible to kip tls verification in helm3? 

Output of `helm version`: v3.1.0

Output of `kubectl version`: v1.17.3

Cloud Provider/Platform (AKS, GKE, Minikube etc.): Self-Hosted

**Comment by hickeyma** [comment]
@mungaij83 There is no Tiller service in Helm 3, so you won't need the TLS capability for it.

This error seems to be saying that you cannot connect to your cluster.

Can you try running the following commands and show the output:
`kubectl cluster-info`
`helm version`

**Comment by mungaij83** [comment]
kubectl cluster-info output:
```
Kubernetes master is running at https://10.2.0.12:8433
KubeDNS is running at https://10.2.0.12:8433/api/v1/namespaces/kube-system/services/kube-dns:dns/proxy

To further debug and diagnose cluster problems, use 'kubectl cluster-info dump'.
```
Helm version output:

`version.BuildInfo{Version:"v3.1.0", GitCommit:"b29d20baf09943e134c2fa5e1e1cab3bf93315fa", GitTreeState:"clean", GoVersion:"go1.13.7"}
`
Note when I do this in the master server, it is okay, but outside the cluster I have to do it this way: `kubectl --insecure-skip-tls-verify=true version`

Since helm lives outside the master server(in production), I cannot make it communicate with kubernetes API server.

**Comment by mungaij83** [comment]
I just pointed out the issue from the clue you gave me. I was looking in the wrong places all the time. 

My kubeconfig file didn't have `certificate-authority-data` for the deployer account. Thanks.

**Comment by wangyp0701** [comment]
helm ls --kube-token ~/token --kube-apiserver http://localhost:8080

**Comment by gmcruz** [comment]
I never would have guessed it but it turns out that Helm 3.2.4 vs. Helm 3.4.1 are very different. It wasn't until I used a totally different vagrant machine with Helm 3.4.1 did it finally work as expected. I spent hours trying to fix this with 3.2.4, hopefully, this helps someone else.

**Comment by Birkenstab** [comment]
For me it helped to explicitly provide the config file location to helm with `--kubeconfig ~/.kube/config`. Even though its the default config path
